config/crd/bases/infrastructure.cluster.x-k8s.io_azuremanagedcontrolplanetemplates.yaml

```diff
@@ -731,6 +731,7 @@ spec:
                           cidrBlock:
                             type: string
                           name:
```

Comment:
Are we sure this PR is actually doing what we want? Isn't the idea that the name here should be removed from the template type?

Reply:
Whoops! This line should refer to the `ClassSpec` type. Will make this change now!

Reply:
Has this change been applied?

Reply:
No. We realized that this would be a breaking API change. The AzureManagedControlPlaneClassSpec references the incorect non-template type instead of the template type here: https://github.com/willie-yao/cluster-api-provider-azure/blob/dont-share-vnet-name/api/v1beta1/types_class.go#L97

We can't fix this due to it being a breaking API change, so we just have a webhook warning for this instead: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4735/files#diff-18af7d1a5d1176aac1e0766c817b6e9d5e079a09e4605ff7c420a4d24d56f85cR43

Discussion about this is here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4735#discussion_r1571256564